landlab/components/dimensionless_discharge/dimensionless_discharge.py

```diff
@@ -0,0 +1,196 @@
+     (lower limits, Tang et al, 2019) have been used, while values of
+     C=0.195 and N=1.27 have been in used in the Italian Dolomites
+     (Gregoretti and Fontana, 2008). Default values are C=12 and N=0.85.
+
+     Parameters
+     ----------
+     soil_density : float list
+         Density of soil in watershed (kg/m^3)
+     water_density : float
+         density of water in waterhed (kg/m^3)
+     C : float
+         Numerator of the debris flow threshold equation; Empirically
+         derived constant (See Tang et al. 2019)
+     N : float
+         Exponent for slope in the denominator of the debris flow
+         threshold equation; Empirically derived constant (See Tang et
+         al. 2019)
+
+     Examples
+     --------
+     >>> from landlab.components import DimensionlessDischarge
+     >>> from landlab import RasterModelGrid
+     >>> import random
+     >>> watershed_grid = RasterModelGrid((3, 3))
+     >>> flux = watershed_grid.add_ones('node', 'flux')
+     >>> d50 = watershed_grid.add_ones('node', 'd50')
+     >>> watershed_grid.at_node ['dem_values'] = \
+     ... np.array([[1.1, 2, 3, 4, 2, 3, 4, 5, 3]])
+     >>> dd = DimensionlessDischarge(watershed_grid)
+     >>> dd.run_one_step()
+     >>> print(watershed_grid.at_node['dimensionless_discharge'])
+     [ 0.55372743  0.55372743  0.55372743  0.55372743  0.55372743
+             0.55372743  0.55372743  0.55372743  0.55372743]
+
+
+
+     References
+     ----------
+     Tang, H., McGuire, L. A., Rengers, F. K., Kean, J. W., Staley,
+     D. M., & Smith, J. B. (2019). Developing and Testing Physically
+     Based Triggering Thresholds for Runoff-Generated Debris Flows.
+     Geophysical Research Letters, 46(15), 8830–8839.
+     https://doi.org/10.1029/2019GL083623
+
+    """
+
+    _name = "DimensionlessDischargeModel"
+
+    _unit_agnostic = False
+
+    _info = {
+        "dimensionless_discharge": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "dimensionless_discharge_above_threshold": {
+            "dtype": bool,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "flux": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m/s",
+            "mapping": "node",
+            "doc": "flux in each stream segment",
+        },
+        "d50": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "soil grain size average in stream segment",
+        },
+        "dem_values": {
```

Comment:
Is this just elevation? Could this name be *"topographic__elevation"*, which is used by lots of other *landlab* components?

Reply:
I think this is topographic elevation. 

I'd also note, that language here is a bit confusing. These are values at node, but the documentation string refers to "stream segment", the most analogous grid element in landlab to a segment is a link.

Reply:
We have specifically been using a digital elevation model (dem) to produce our elevation values, but topographic__elevation likely captures the information that is needed better. I will make that change.

@kbarnhart I have been a bit confused on the best grid types to use for the different variables (I have little experience working with grids), so if you think a link would be better I can make that update. Stream segments are sections of the stream that we average over as we can not run this model over every single point along a river.

Reply:
After thinking about thinks in the context of this problem I think that using nodes for this would still be best. We need a slope for each stream segment, so an elevation is given for each location of a stream segment (our nodes). I am not sure how you would convert links into elevations for each node.

Reply:
I think what you have is fine.